aesara/graph/opt.py

```diff
@@ -1692,9 +1692,6 @@ def transform(self, fgraph, node, get_nodes=True):
                     continue
                 ret = self.transform(fgraph, real_node, get_nodes=False)
                 if ret is not False and ret is not None:
-                    assert len(real_node.outputs) == len(ret)
```

Comment:
This checks don't make sense here. There is no reason why it should be treated differently than the normal `self.transform` case